lib/tapioca/rbi/printer.rb

```diff
@@ -0,0 +1,406 @@
+# typed: strict
+# frozen_string_literal: true
+
+module Tapioca
+  module RBI
+    class Printer < Visitor
+      extend T::Sig
+
+      sig { returns(T::Boolean) }
+      attr_accessor :in_visibility_group
+
+      sig { returns(T.nilable(Node)) }
+      attr_reader :previous_node
+
+      sig { params(out: T.any(IO, StringIO), indent: Integer).void }
+      def initialize(out: $stdout, indent: 0)
+        super()
+        @out = out
+        @current_indent = indent
+        @in_visibility_group = T.let(false, T::Boolean)
+        @previous_node = T.let(nil, T.nilable(Node))
+      end
+
+      # Printing
+
+      sig { void }
+      def indent
+        @current_indent += 2
+      end
+
+      sig { void }
+      def dedent
+        @current_indent -= 2
+      end
+
+      sig { params(string: String).void }
+      def print(string)
+        @out.print(string)
+      end
+
+      sig { params(string: T.nilable(String)).void }
+      def printn(string = nil)
+        print(string) if string
+        print("\n")
+      end
+
+      sig { params(string: T.nilable(String)).void }
+      def printt(string = nil)
```

Comment:
Can we use a more descriptive name? Maybe `indented_print`?

Reply:
I would prefer not.

I chose those names so it aligns nicely in the code:

```rb
v.indent
v.printt "some code"
v.printn "some other code"
v.printl "more code"
v.dedent
```

I added a comment on each of them to explain the behaviour. Is it better?